### PR TITLE
Wikipedia training

### PR DESCRIPTION
**Pull request by micmelesse**
This PR adds scripts for downloading and preprocessing wikipedia, training on wikipedia on a single node and extracting model metrics over time such as the loss.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,6 +65,24 @@ the `--optimizer_type` commandline option set to the following:
...
 
+### E. Wikipedia Training
+We can use Horovod to train on a single node using the following scripts. 
+
+First download and preprocess wikipedia with the following script. By default it preprocess for sequence length 128 but you can alter that inside the script.
+```
+bash scripts/get_wikipedia_dataset.sh
+```
+
+Then run pretraining inside our container. By default it assume an 8 GPU node with 1000000 steps but you can adjust the number of GPUs and number of steps inside the script. Note that this will note give you a fully trained model. The original paper trained on 64 TPUs for 1000000 steps.
```
Comment (c0redumb) [inline README.md:76]: Typo: note => not

**Comment by c0redumb** [comment]
Michael, I was not sure if this should go into the repos as part of the code, or go into the confluence as a part of wiki. This needs some time to think through.

For example, in the other scripts/run*.sh files, I have assumed that the host is a bare metal machine. With no TensorFlow or even Python installed, everything is run inside docker containers. This avoids dependency complexity and instructions on what to install. This also makes it possible for people who just wanted to try our code without messing their host setup. Remember that many of our audiences are outside of our organization.

This instruction as we described here are mostly intended for internal consumption. So it is probably best to be left on confluence? Maybe just to ask people to copy already processed Wikipedia from our server, and modify the scripts to run with NP=8 and data from Wikipedia? If we want this code to live on the repos, we may have to modify all the script (preprocessing, metric extraction) to be run in containers, and also remember that the container released does not have tensorboard which your code depends on.

**Comment by micmelesse** [comment]
@c0redumb We are going to have to eventually add scripts for Wikipedia training so I can create the instructions in Confluence and also work on making the preprocessing and metric extraction run in containers.

**Comment by c0redumb** [comment]
That would be a good plan.

Another suggestion: Since there are a lot of common codes among the scripts (such as launching the container, sort through the folders inside and outside, etc), maybe it helps to think of a way to put this at one place and sourced by other scripts (instead of having it duplicated in every script). The idea is to take a bare metal host and have the scripts working.

Thanks.

**Comment by MikalaiDrabovich** [comment]
from https://github.com/google-research/bert#pre-training-data:
there is a step to:
 "apply any necessary cleanup to convert it into plain text"

For example for Megatron-LM,  nltk is used for this:
https://github.com/NVIDIA/Megatron-LM/blob/master/tools/preprocess_data.py#L66

Is there any equivalent step for TF data being applied? 
Otherwise, it may cause lower model  accuracy/ppl after training, not attributed to AMD ecosystem.
